src/backend/distributed/progress/multi_progress.c

```diff
@@ -133,6 +133,17 @@ FinalizeCurrentProgressMonitor(void)
 }
 
 
+/*
+ * GetCurrentProgressDSMHandle is the getter function for static variable
+ * currentProgressDSMHandle
+ */
+uint64
+GetCurrentProgressDSMHandle(void)
```

Comment:
I think the result of this function is a bit too much of an implementation detail. Two options that I think would be better:
1. Create `HasProgressMonitor()` which returns `bool`
2. Return NULL from `GetCurrentProgressMonitor()` if there's no registered monitor.